addons/zookeeper/scripts/startup.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Execute entrypoint as usual after obtaining ZOO_SERVER_ID
+# check ZOO_SERVER_ID in persistent volume via myid
+# if not present, set based on POD hostname
+# Execute entrypoint as usual after obtaining ZOO_SERVER_ID
+# check ZOO_SERVER_ID in persistent volume via myid
+# if not present, set based on POD hostname
+if [[ -f "/bitnami/zookeeper/data/myid" ]]; then
+  export ZOO_SERVER_ID="$(cat /bitnami/zookeeper/data/myid)"
+else
+  SERVICE_ID=$((${KB_POD_NAME##*-}))
```

Comment:
Does SERVICE_ID necessarily need to be of type int? Obtaining SERVICE_ID in this way may cause conflicts in scenarios involving instanceTemplate.

Reply:
Furthermore, try to avoid using any built-in env with the prefix KB_. All env should be explicitly declared in vars or env. The built-in env with the prefix KB_ will be directly removed in subsequent 1.0 versions. If you need POD_NAME, you should use the native way to obtain it. For example:

```
          - name: CURRENT_POD_NAME
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: metadata.name
```

Reply:
done